### PR TITLE
feat(secrets): add scm host metadata

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HAC-5715
https://issues.redhat.com/browse/HAC-5717

## Description

- Adds input field for host & repo in AddSecret form.
- removes remote secret labels from Secret creation utils.
- For source secret adds scm labels and annotation in createSecret utils.
- Modifies and adds to related unit tests
- Updates Secret List View & related utils to use K8sSecrets instead of Remote Secrets.
- Links & unlinks Image pull Secrets to Service accounts on creation and deletion

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

scm-secrets

Screenshot 2024-05-14 at 4 48 00 PM
Screenshot 2024-05-15 at 5 18 04 PM
Screenshot 2024-05-15 at 6 25 50 PM

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by sahil143 — CHANGES_REQUESTED** [review]
@abhinandan13jan Please remove the creation of RemoteSecret. We will only support k8s Secrets.

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/components/Secrets/secret-modal.tsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+import { unLinkSecretFromServiceAccount } from './utils/service-account-utils';
 
-export const secretDeleteModal = (secret: RemoteSecretKind) =>
+export const secretDeleteModal = (secret: SecretKind) =>
   createDeleteModalLauncher(secret.kind)({
     obj: secret,
-    model: RemoteSecretModel,
+    model: SecretModel,
+    submitCallback:
+      secret.type === SecretType.dockerconfigjson ? unLinkSecretFromServiceAccount : null,
```
Comment (sahil143) [inline src/components/Secrets/secret-modal.tsx:12]: why use a conditional check here to pass the callback?

File: src/components/Secrets/utils/secret-utils.ts [hunk 2]
```diff
@@ -236,55 +251,58 @@ export const createSecretResource = async (
...
-        type: secret.type,
-        ...(labels?.secret && { labels: labels.secret }),
-      },
-      targets: [{ namespace }],
-    },
-  };
+// export const createSecretResourceWithMetadata = (
+//   secret: SecretKind,
+//   namespace: string,
+//   labels: {
```
Comment (sahil143) [inline src/components/Secrets/utils/secret-utils.ts:257]: remove commented code

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/components/Secrets/secret-modal.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply src/components/Secrets/secret-modal.tsx:12]: linking/unlinking for imagePull Secret only, other types this callback not needed

**Review by sahil143 — CHANGES_REQUESTED** [review]
Inconsistent styling

image

**Comment by abhinandan13jan** [comment]
> Inconsistent styling
> 
> image

@sahil143 I dint change the CSS, the image you shared is what the existing form looked like. 
I've added a commit to reduce the sizing of the dropdown fields to align with that of our standard input fields. However, the form looks different in 3 different SecretType cases

Screenshot 2024-05-20 at 5 47 31 PM
Screenshot 2024-05-20 at 5 47 44 PM
Screenshot 2024-05-20 at 5 48 00 PM

**Comment by pankajshivpuje** [comment]
@abhinandan13jan I will get a quick microcopy review done by the docs team. I am also assuming that the Host and Repository fields are optional, correct?

**Comment by karthikjeeyar** [comment]
Secret Type column must map to `Key/Value (#)` or `image`, now it shows some random label values.
 
image

**Comment by abhinandan13jan** [comment]
> @abhinandan13jan I will get a quick microcopy review done by the docs team. I am also assuming that the Host and Repository fields are optional, correct?

@pankajshivpuje yes, these are optional

**Comment by karthikjeeyar** [comment]
@pankajshivpuje @abhinandan13jan  Since deployment secrets are removed, do we want to update learn more content as well? It still contains deployment secrets section in it. 

image

**Comment by rohitkrai03** [comment]
@karthikjeeyar We are going to replace learn more links with direct links to documentation because they do not work in community distribution installation on Konflux.

**Comment by abhinandan13jan** [comment]
> Secret Type column must map to `Key/Value (#)` or `image`, now it shows some random label values.
> 
> image

@karthikjeeyar there are several other types, only these 3 
    case SecretType.basicAuth:
    case SecretType.sshAuth:
    case SecretType.opaque:
      return SecretTypeDisplayLabel.keyValue; are part of KeyValue, 
  We have kubernetes/service-tokewn and managed types. The util for showing types has not been changes.
  Perhaps we are noticing this change because we have switched from `RemoteSecrets` list to `SecretsList`. In case we only want to show a filtered list of items with type `keyvalue` and `image-pull` will be a different use case? WDYT?

**Comment by abhinandan13jan** [comment]
> Creating a pull secret with dockerconfig auth type is not linking the created secret to `appstudio-pipeline` SA.
> 
> image
> 
> SA not linked to `image-pull-secret-test` image

Updated. Can you please check??

**Comment by karthikjeeyar** [comment]
>  In case we only want to show a filtered list of items with type keyvalue and image-pull will be a different use case? WDYT?

 I see these secrets with  type `managed-gitops.redhat.com/managed-environment` in cli as well.  These secrets might not be user created ones, but lets get this in and based on feedback, we revisit and filter it need.

**Review by karthikjeeyar — APPROVED** [review]

**Comment by pankajshivpuje** [comment]
> @pankajshivpuje @abhinandan13jan Since deployment secrets are removed, do we want to update learn more content as well? If yes I think we need to notify the docs team about this.
> 
> image

@karthikjeeyar I will request support from docs team to have this updated. I will let you know when I hear back from them.

**Comment by pankajshivpuje** [comment]
@abhinandan13jan My apologies for commenting late for this PR. Just wanted to check the support text for Repository field. Is this the repository where secret will be stored or is this the repository where this secret will be used?

**Pull request merged**